File: backend/src/main/java/com/project2/domain/post/entity/PostImage.java
package com.project2.domain.post.entity;

import com.project2.global.entity.BaseTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Entity
public class PostImage extends BaseTime {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@Column(nullable = false, length = 500)
	private String imageUrl;

	@ManyToOne
	@JoinColumn(nullable = false)
	private Post post;
}

File: backend/src/main/kotlin/com/project2/domain/post/entity/PostImage.kt
package com.project2.domain.post.entity

import com.project2.global.entity.BaseTime
import jakarta.persistence.*

@Entity
class PostImage : BaseTime() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null

    @Column(nullable = false, length = 500)
    lateinit var imageUrl: String

    @ManyToOne
    @JoinColumn(nullable = false)
    lateinit var post: Post
}